variational_longitudinal_rate_of_aging_autoencoder.py

```diff
@@ -72,50 +71,59 @@ def _train_epoch(self, regularization_weighting):
         longitudinal_ages0 = train_longitudinal_ages0[perm]
         longitudinal_ages1 = train_longitudinal_ages1[perm]
         
-        # make sure we have the same number of train longitudinal batches and cross-sectional batches. 
-        # TODO: is this really what we want to do? Longitudinal batches are much smaller, introducing noise? 
-        train_longitudinal_batches = np.array_split(range(n_longitudinal_points), n_train_batches)
-        assert len(train_longitudinal_batches) == n_train_batches
-        assert sorted(list([a for b in train_longitudinal_batches for a in b])) == list(range(n_longitudinal_points))
-        
-        # train, alternating between minimizing cross-sectional and longitudinal loss. 
+        # create longitudinal batches
+        train_longitudinal_batches = divide_idxs_into_batches(
+            np.arange(n_longitudinal_points), 
+            self.longitudinal_batch_size)
+        n_longitudinal_batches = len(train_longitudinal_batches)
+        
+        print(("%i cross-sectional batches of (approximate) size %i; "+
+              "%i longitudinal batches of size %i") % (n_cross_sectional_batches,
+                                                       self.batch_size,
+                                                       n_longitudinal_batches, 
+                                                       self.longitudinal_batch_size))
+        
+        # generate a vector that indicates whether we're doing a longitudinal or 
+        # cross-sectional update at each step. 
+        batch_is_longitudinal = ([False for i in range(n_cross_sectional_batches)] + 
+                                 [True for i in range(n_longitudinal_batches)])
+        np.random.shuffle(batch_is_longitudinal)
```

Comment:
I think we want a longitudinal batch size parameter because we may need it to control noise on the gradient update. If we're worried about washing out the longitudinal updates, we can just reduce the gradient batch size and/or increase the weighting? Either way, it seems like we want two parameters to control both noise + relative longitudinal weighting.